mlflow/server/js/src/components/RunView.js

```diff
@@ -98,8 +125,45 @@ class RunView extends Component {
     this.setState({ showRunRenameModal: false });
   }
 
+  renderNoteSection(noteInfo) {
+    if (this.state.showNotes) {
```

Comment:
We should make sure that the expander can not be clicked when the notes editor is being shown. In the current implementation, if we close the expander, the state in the `NoteEditorView` will be lost since we are unmounting that component.

Likewise, we should make sure that any route changes are protected with a Prompt. I believe you can do this with this component. https://reacttraining.com/react-router/core/api/Prompt

Reply:
I tried making it so that RunView would store the state of the editor whenever it was going to get unmounted and then restore it on next render, but that got really hairy without using refs.

Should the `onclick` on the expander just be disabled and the caret hidden when the editor is shown? That feels a bit counterintuitive, but I don't have a better, simple idea.

Reply:
Yeah, let's just disable the onClick on the expander.

Reply:
Done